minidump-processor/src/op_analysis.rs

```diff
@@ -0,0 +1,733 @@
+//! Module for analyzing CPU instructions
+//!
+//! When analyzing a minidump, it is often useful to know information about what instructions
+//! were being run by various CPU threads (especially the crashing thread during a crash)
+//!
+//! This module attempts to provide a toolbox of instruction analysis tools that can be used to
+//! provide such information
+//!
+//! Support for different architectures can be enabled through features on the crate. Below is
+//! a list of currently available architectures and enabling features:
+//!
+//! - `disasm_amd64`: enable analysis of Amd64 instructions (on by default)
+//!
+//! The functions in this module will generally return `OpAnalysisError::UnsupportedCpuArch` if
+//! support for the target CPU is not available
+//!
+//!
+
+#![deny(missing_docs)]
+
+use minidump::{MinidumpContext, MinidumpRawContext};
+use std::io::{Read, Seek, SeekFrom};
+
+/// The maximum size of an instruction on any supported architecture (currently Amd64)
+///
+/// For architectures that have variable-length instructions (such as x86 family), filling a
+/// buffer with this many bytes guarantees that decode will never fail due to instruction
+/// truncation
+///
+/// Note that the opposite is explicitly not true -- filling a buffer with fewer
+/// bytes than this does not guarantee that decoding will fail. In some cases, only a single
+/// byte may be needed to represent an instruction
+///
+/// Specifying too low of a value for this may cause long instructions to fail to decode. Too
+/// high of a value won't affect functionality, but may result in unnecessary memory usage
+pub const MAX_INSTRUCTION_LENGTH: u8 = MAX_AMD64_INSTRUCTION_LENGTH;
+
+/// The maximum size of an instruction on Amd64 architecture
+///
+/// Amd64 instructions can be between 1 and 15 bytes in length
+pub const MAX_AMD64_INSTRUCTION_LENGTH: u8 = 15;
```

Comment:
There is a reason for this -- Currently `std::io::Read::take()` uses a `u64` as its argument, and since `From<usize>` is not implemented for `u64` it means I either have to `as` cast or use `TryFrom`, neither of which I love considering I know statically that the value is <256